pkg/statistics/handle/handle.go

```diff
@@ -204,6 +204,34 @@ func (h *Handle) getPartitionStats(tblInfo *model.TableInfo, pid int64, returnPs
 	return tbl
 }
 
+// GetPartitionStatsByID retrieves the partition stats from cache by partition ID.
+func (h *Handle) GetPartitionStatsByID(is infoschema.InfoSchema, pid int64) *statistics.Table {
+	return h.getPartitionStatsByID(is, pid, true)
+}
+
+func (h *Handle) getPartitionStatsByID(is infoschema.InfoSchema, pid int64, returnPseudo bool) *statistics.Table {
```

Comment:
Why do we need an extra `returnPseudo` if it can only be `true`?

Reply:
It's just copied from the original function. Not begin to refactor it in the first commit(